my-dsl.md

```diff
@@ -2,37 +2,79 @@
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+AWK was originally designed to process text and manipulate data. Thus, it is 
+mostly used in one-liner scripts/programs that do these domain-specific things. 
+However, since AWK is Turing-complete, it can be used like a general-
+purpose language and has been used to create large AWK programs. 
+
+On the spectrum of languages, it sits between general-purpose and purely domain-
+specific languages, leaning towards purely domain-specific. This is because its 
+intended use was for text manipulation, and it is most often used in one-liner 
+scripts inside larger applications or as part of a programmer's workflow. 
+However, as mentioned above, it is Turing-Complete and it is capable of doing 
+much more than purely text processing and data extraction, so it can be used as 
+a general-purpose language. Fowler argues that languages that are Turing-
+complete should not be considered DSLs. However, Fowler also argues that the 
+way a language is used also matters. AWK is mostly used to do text processing, 
+so it is definitely more on the domain-specific side of things.
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+It is an external DSL. AWK is usually used as small scripts in larger programs 
+that use languages other than AWK. Since Fowler defines external DSLs as 
+a language separate from the main language of the application it works with, 
+AWK fits the bill.
+
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+It was written in C.
 
 # Benefits
 _Identify one potential benefit of the DSL: how is a programmer's life or a 
 company's bottom line made easier by the existence of this language?_
 
+If a programmer is fluent in AWK, some interesting text manipulations or data 
+extractions can be very easy. AWK was used widely to put together reports, so 
+used in those domains AWK can save the programmer and their company time and 
+therefore money. Also, since AWK is widely available on UNIX distributions, 
+it saves even more time because people do not need to waste time downloading 
+installing software.
 
 # Drawbacks
 _Identify one potential drawback of the DSL: what does a programmer or company 
 lose by using this DSL instead of a general-purpose language?_
+
+AWK is an old language and largely succeeded by Perl and other scripting 
+languages. That is because AWK has a very specialized and different syntax that 
+is just not worth the time and effort to learn and memorize. So a potential 
+drawback of using the DSL is lack of results from learning and knowing it. If 
+a programmer is pretty good with Perl or Python, they can do relatively the 
+same things AWK can in relatively the same time. In that case, learning AWK is 
+a waste of time in terms of productivity.
```

Comment:
I agree. This seems to be a drawback for a lot of DSLs, in my opinion. It is definitely time-consuming to learn new syntax and rules of some DSLs.